From 542db56eddba85538fbbb9c65eaeff84996246eb Mon Sep 17 00:00:00 2001
From: Jakob Kastelic <kastelic.jakob@gmail.com>
Date: Thu, 18 Sep 2025 20:38:22 -0700
Subject: [PATCH 1/1] allow loading linux kernel as BL33

Increase the SD card reading timeout to allow for a larger read,
increase maximum BL33 size, and load the BL33 further into the memory
space to avoid the need for relocation.

Signed-off-by: Jakob Kastelic <kastelic.jakob@gmail.com>
---
 drivers/st/mmc/stm32_sdmmc2.c   | 2 +-
 plat/st/stm32mp1/stm32mp1_def.h | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/st/mmc/stm32_sdmmc2.c b/drivers/st/mmc/stm32_sdmmc2.c
index be722f322..62530fe25 100644
--- a/drivers/st/mmc/stm32_sdmmc2.c
+++ b/drivers/st/mmc/stm32_sdmmc2.c
@@ -616,7 +616,7 @@ static int stm32_sdmmc2_read(int lba, uintptr_t buf, size_t size)
 		flags |= SDMMC_STAR_DBCKEND;
 	}
 
-	timeout = timeout_init_us(TIMEOUT_US_1_S);
+	timeout = timeout_init_us(TIMEOUT_US_1_S * 5);
 
 	do {
 		status = mmio_read_32(base + SDMMC_STAR);
diff --git a/plat/st/stm32mp1/stm32mp1_def.h b/plat/st/stm32mp1/stm32mp1_def.h
index 6530957c3..a8d8c7406 100644
--- a/plat/st/stm32mp1/stm32mp1_def.h
+++ b/plat/st/stm32mp1/stm32mp1_def.h
@@ -179,12 +179,12 @@ enum ddr_type {
 #endif
 
 #if STM32MP13
-#define STM32MP_BL33_BASE		STM32MP_DDR_BASE
+#define STM32MP_BL33_BASE		(STM32MP_DDR_BASE + U(0x2008000))
 #endif
 #if STM32MP15
 #define STM32MP_BL33_BASE		(STM32MP_DDR_BASE + U(0x100000))
 #endif
-#define STM32MP_BL33_MAX_SIZE		U(0x400000)
+#define STM32MP_BL33_MAX_SIZE		U(0x3FF8000)
 
 /* Define maximum page size for NAND devices */
 #define PLATFORM_MTD_MAX_PAGE_SIZE	U(0x1000)
-- 
2.39.5

